### DISCUSSION TITLE
Wouldn't it be useful to be able to clear the current context?

### DISCUSSION THREAD

**Discussion by MasayukiOzawa** [microsoft/go-sqlcmd#370] [Ideas]
When a SQL Server container is created with sqlcmd create, a definition for the automatically created environment is added to sqlconfig and the currentcontext is set.
When connecting to the created environment / using sqlconfig, the information registered in sqlconfig is very useful.

However, if you want to run sqlcmd as a conventional use of sqlcmd, the inclusion of the currentcontext information will result in a different feel for specifying user information than in the past.

If sqlconfig exists with currentcontext set, the password setting seems to override the context password, and setting SQLCMDPASSWORD is ignored.

If currentcontext is set, the following command will result in unexpected login behavior.
```
$ENV:SQLCMDPASSWORD=xxxxxx
sqlcmd -S xxxxx -U xxxxxx
```
The expected logon is with the password for $ENV:SQLCMDPASSWORD, but if currentsqlcontext exists in sqlconfig, the logon is done with the password for the specified context.

If the currentcontext setting can be cleared with a command such as `sqlcmd config clear-current-context`, then even if sqlconfig is set, logins will be the same as before sqlconfig was set.

**Comment by apoorvdeshmukh** [comment]
This is happening because we expect username and password to come together either from environment variables or from sqlconfig file as mentioned[ here](https://github.com/microsoft/go-sqlcmd/blob/9d974bb9703d91b7adb8f6c3839844e5eeab649e/cmd/modern/main.go#LL88C2-L88C2).
Can you also try setting username also via env variable SQLCMDUSER? It should work.

**Reply by MasayukiOzawa** [reply]
I see, so specifying SQLCMDUSER will result in the conventional behavior.
If you can do that behavior, there's no need to explicitly clear it.

It would be nice to see a description of this behavior in the documentation.
